src/components/ripple/ripple.ts

```diff
@@ -0,0 +1,288 @@
+      if (this._triggerElement) {
+        this._triggerElement.addEventListener('mousedown', this._mouseDownHandler);
+        this._triggerElement.addEventListener('click', this._clickHandler);
+        this._triggerElement.addEventListener('mouseleave', this._mouseLeaveHandler);
+      }
+    }
+  }
+
+  /**
+   * Responds to the start of a ripple animation trigger by fading the background in.
+   */
+  start() {
+    this._rippleManager.showRippleBackground(this._rippleBackground, this._backgroundColor);
+  }
+
+  /**
+   * Responds to the end of a ripple animation trigger by fading the background out, and creating a
+   * foreground ripple that expands from the event location (or from the center of the element if
+   * the "centered" property is set or forceCenter is true).
+   */
+  end(left: number, top: number, forceCenter = true) {
+    this._rippleManager.createForegroundRipple(
+      this._element,
+      left,
+      top,
+      this._color,
+      this._centered || forceCenter,
+      this._speedFactor,
+      (ripple: Ripple, event: TransitionEvent) => this._rippleTransitionEnded(ripple, event));
+    // Fade out the highlighted background.
+    this._rippleManager.hideRippleBackground(this._rippleBackground);
+  }
+
+  private _rippleTransitionEnded(ripple: Ripple, event: TransitionEvent) {
+    if (event.propertyName === 'opacity') {
+      // If the ripple finished expanding, start fading it out. If it finished fading out,
+      // remove it from the DOM.
+      switch (ripple.state) {
+        case RippleState.EXPANDING:
+          this._rippleManager.fadeOutForegroundRipple(ripple.element);
+          ripple.state = RippleState.FADING_OUT;
+          break;
+        case RippleState.FADING_OUT:
+          this._rippleManager.removeRippleFromDom(ripple.element);
+          break;
+      }
+    }
+  }
+
+  mouseDown(event: MouseEvent) {
+    if (!this._disabled && event.button === 0) {
+      this.start();
+    }
+  }
+
+  click(event: MouseEvent) {
+    if (!this._disabled && event.button === 0) {
+      // If screen and page positions are all 0, this was probably triggered by a keypress.
+      // In that case, use the center of the bounding rect as the ripple origin.
+      // FIXME: This fails on IE11, which still sets pageX/Y and screenX/Y on keyboard clicks.
+      const isKeyEvent =
+          (event.screenX === 0 && event.screenY === 0 && event.pageX === 0 && event.pageY === 0);
+      this.end(event.pageX, event.pageY, isKeyEvent);
+    }
+  }
+
+  mouseLeave(event: MouseEvent) {
+    // We can always fade out the background here; It's a no-op if it was already inactive.
+    this._rippleManager.hideRippleBackground(this._rippleBackground);
+  }
+
+  // TODO: Reactivate the background div if the user drags out and back in.
+}
+
+/**
+ * Helper service that performs DOM manipulations. Not intended to be used outside this module.
+ */
+class MdInkRippleManager {
+  constructor(private _renderer: Renderer) {}
```

Comment:
You shouldn't need to go through the existing `Renderer` in this class since we're already doing a bunch of direct DOM stuff outside of it